publication-rest/src/main/java/no/unit/nva/publication/create/CreatePublicationFromImportCandidateHandler.java

```diff
@@ -72,16 +82,31 @@ protected PublicationResponse processInput(ImportCandidate input, RequestInfo re
                    .map(candidateService::getImportCandidateByIdentifier)
                    .map(candidate -> injectOrganizationAndOwner(requestInfo, candidate))
                    .map(publicationService::autoImportPublication)
+                   .map(this::copyArtifacts)
                    .map(CreatePublicationFromImportCandidateHandler::toPublicationUriIdentifier)
-                   .map(identifier -> candidateService.updateImportStatus(input.getIdentifier(), toImportStatus(requestInfo, identifier)))
-                   .map(importCandidate -> publicationService.getPublicationByIdentifier(extractPublicationId(importCandidate)))
+                   .map(identifier -> candidateService.updateImportStatus(input.getIdentifier(),
+                                                                          toImportStatus(requestInfo, identifier)))
+                   .map(importCandidate -> publicationService.getPublicationByIdentifier(
+                       extractPublicationId(importCandidate)))
                    .map(PublicationResponse::fromPublication)
                    .orElseThrow(failure -> rollbackAndThrowException(input));
     }
 
+    private Publication copyArtifacts(Publication publication) {
+        publication.getAssociatedArtifacts().stream()
+            .filter(File.class::isInstance)
+            .forEach(a -> s3.copyObject(CopyObjectRequest.builder()
```

Comment:
I would extract s3.copy... to a new method, something like:

`file -> copyFileToPersistedStorage(file)`